### PR TITLE
Dump reconstruction method

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/DgSubcell/Tags/MethodOrder.hpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+struct MethodOrder : db::SimpleTag {
+  using type = std::optional<tnsr::I<DataVector, Dim, Frame::ElementLogical>>;
+};
+
+template <size_t Dim>
+struct MethodOrderCompute : db::ComputeTag, MethodOrder<Dim> {
+  using base = MethodOrder<Dim>;
+  using return_type = typename base::type;
+  using argument_tags =
+      tmpl::list<::domain::Tags::Mesh<Dim>, Mesh<Dim>, ActiveGrid,
```
Comment (kidder) [inline src/Evolution/DgSubcell/Tags/MethodOrder.hpp:46]: include for tmpl

File: src/Evolution/DgSubcell/Tags/MethodOrder.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  } else {
+    destructive_resize_components(make_not_null(&method_order->value()),
+                                  subcell_mesh.number_of_grid_points());
+    if (reconstruction_order.has_value()) {
+      *method_order = reconstruction_order;
+    } else {
+      if (UNLIKELY(static_cast<int>(
+                       subcell_options.finite_difference_derivative_order()) <
+                   0)) {
+        ERROR(
```
Comment (kidder) [inline src/Evolution/DgSubcell/Tags/MethodOrder.cpp:40]: include for ERROR

File: src/Evolution/DgSubcell/ReconstructionOrder.hpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            for (size_t j = 0;
+                 j < subcell_mesh.slice_away(d).number_of_grid_points(); ++j) {
+              for (size_t i = 0; i < subcell_mesh.extents(d); ++i) {
+                get<0>(recons_order_ptr
+                           ->value())[j * subcell_mesh.extents(d) + i] =
+                    gsl::at(reconstruction_order.value(),
+                            d)[j * (subcell_mesh.extents(d) + 2) + i + 1];
+              }
+            }
+            // if (d == 0) {
```
Comment (kidder) [inline src/Evolution/DgSubcell/ReconstructionOrder.hpp]: remove

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

**Comment by nilsdeppe** [comment]
Squashed and rebased. Thanks for the review!

**Review by kidder — APPROVED** [review]

**Pull request merged**